statsmodels/duration/survfunc.py

```diff
@@ -0,0 +1,331 @@
+    d = d[ii]
+    n = n[ii]
+    time = time[ii]
+
+    # The survival function probabilities.
+    sp = 1 - d / n.astype(np.float64)
+    sp = np.log(sp)
+    sp = np.cumsum(sp)
+    sp = np.exp(sp)
+
+    # Standard errors (Greenwood's formula).
+    se = d / (n * (n - d)).astype(np.float64)
+    se = np.cumsum(se)
+    se = np.sqrt(se)
+    se *= sp
+
+    return sp, se, time, n, d
+
+
+
+class survfunc_right(object):
+    """
+    Estimation and inference for a survival function.
+
+    Only right censoring is supported.
+
+    Parameters
+    ----------
+    time : array-like
+        An array of times (censoring times or event times)
+    status : array-like
+        Status at the event time, status==1 is the 'event'
+        (e.g. death, failure), meaning that the event
+        occurs at the given value in `time`; status==0
+        indicates that censoring has occured, meaning that
+        the event occurs after the given value in `time`.
+    title : string
+        Optional title used for plots and summary output.
+
+    Attributes
+    ----------
+    surv_prob : array-like
+        The estimated value of the survivor function at each time
+        point in `surv_times`.
+    surv_prob_se : array-like
+        The standard errors for the values in `surv_prob`.
+    surv_times : array-like
+        The points where the survival function changes.
+    n_risk : array-like
+        The number of subjects at risk just before each time value in
+        `surv_times`.
+    n_events : array-like
+        The number of events (e.g. deaths) that occur at each point
+        in `surv_times`.
+    """
+
+    def __init__(self, time, status, title=None):
+
+        self.time = time
+        self.status = status
+        m = len(status)
+        x = _calc_survfunc_right(time, status)
+        self.surv_prob = x[0]
+        self.surv_prob_se = x[1]
+        self.surv_times = x[2]
+        self.n_risk = x[3]
+        self.n_events = x[4]
+        self.title = "" if not title else title
+
+
+    def plot(self, ax=None):
+        """
+        Plot the survival function.
+
+        Examples
+        --------
+        Change the line color:
+
+        >>> fig = sf.plot()
+        >>> ax = fig.get_axes()[0]
+        >>> ha, lb = ax.get_legend_handles_labels()
+        >>> ha[0].set_color('purple')
+        >>> ha[1].set_color('purple')
+
+        Don't show the censoring points:
+
+        >>> fig = sf.plot()
+        >>> ax = fig.get_axes()[0]
+        >>> ha, lb = ax.get_legend_handles_labels()
+        >>> ha[1].set_visible(False)
+        """
+
+        return plot_survfunc(self, ax)
+
+
+    def quantile(self, p):
+        """
+        Estimated quantile of a survival distribution.
+
+        Parameters
+        ----------
+        p : float
+            The probability point at which the quantile
+            is determined.
+
+        Returns the estimated quantile.
+        """
+
+        ii = np.flatnonzero(self.surv_prob < 1 - p)
```

Comment:
I got this from here, where they use a strict inequality:

http://support.sas.com/documentation/cdl/en/statug/63033/HTML/default/viewer.htm#statug_lifetest_sect004.htm